accounts/accounts_cli.go

```diff
@@ -0,0 +1,328 @@
+}
+
+// Account details that we will audit
+type Account struct {
+	Id      string `json:"userid"`
+	Profile struct {
+		FullName string `json:"fullName"`
+		Patient  struct {
+			Bday string `json:"birthday"`
+			Dday string `json:"diagnosisDate"`
+		} `json:"patient"`
+	} `json:"patientProfile"`
+	Perms      interface{} `json:"permissons"`
+	LastUpload string      `json:"lastupload"`
+}
+
+var (
+	prodHost    = "https://api.tidepool.io"
+	stagingHost = "https://staging-api.tidepool.io"
+	develHost   = "https://devel-api.tidepool.io"
+	localHost   = "http://localhost:8009"
+)
+
+func main() {
+
+	app := cli.NewApp()
+
+	app.Name = "Accounts-Managment"
+	app.Usage = "Allows the bulk management of tidepool accounts"
+	app.Version = "0.0.1"
+	app.Author = "Jamie Bate"
+	app.Email = "jamie@tidepool.org"
+
+	app.Commands = []cli.Command{
+
+		//e.g. audit -u admin@place.org -e prod
+		//e.g. audit -u admin@place.org -e prod -r ./audit_admin@place.org_accounts_prod_2015-07-01T03:00:58Z.txt
+		{
+			Name:      "audit",
+			ShortName: "a",
+			Usage:     `audit all accounts that you have permisson to access`,
+			Flags: []cli.Flag{
+				cli.StringFlag{
+					Name:  "e, env",
+					Value: "local",
+					Usage: "the environment your running against. Options are local, devel, staging and prod",
+				},
+				cli.StringFlag{
+					Name:  "u, username",
+					Usage: "your tidepool username that has access to the accounts e.g. admin@tidepool.org",
+				},
+				cli.StringFlag{
+					Name:  "r, reportpath",
+					Usage: "rerun the process on an existing report that you have given the path too e.g -r ./accountsAudit_2015-06-26 08:17:29.445414108 +1200 NZST.txt",
+				},
+			},
+			Action: auditAccounts,
+		},
+	}
+
+	app.Run(os.Args)
+
+}
+
+func (a *admin) findPublicInfo(acc *Account) error {
+
+	urlPath := a.host + fmt.Sprintf("/metadata/%s/profile", acc.Id)
+
+	req, _ := http.NewRequest("GET", urlPath, nil)
+	req.Header.Add("x-tidepool-session-token", a.User.Token)
+
+	res, err := a.client.Do(req)
+
+	if err != nil {
+		return errors.New("Could attempt to find profile: " + err.Error())
+	}
+
+	switch res.StatusCode {
+	case 200:
+		data, _ := ioutil.ReadAll(res.Body)
+		json.Unmarshal(data, &acc.Profile)
+		return nil
+	default:
+		log.Printf("Failed finding public info [%d] for [%s]", res.StatusCode, acc.Id)
+		return nil
+	}
+}
+
+func (a *admin) findLastUploaded(acc *Account) error {
+
+	urlPath := a.host + fmt.Sprintf("/query/upload/lastentry/%s", acc.Id)
+
+	req, _ := http.NewRequest("GET", urlPath, nil)
+	req.Header.Add("x-tidepool-session-token", a.User.Token)
+
+	res, err := a.client.Do(req)
+
+	if err != nil {
+		return errors.New("Could attempt to find the last upload: " + err.Error())
+	}
+
+	switch res.StatusCode {
+	case 200:
+		data, err := ioutil.ReadAll(res.Body)
+
+		if err != nil {
+			log.Println("Error trying to read the last upload data", err.Error())
+			return nil
+		}
+
+		json.Unmarshal(data, &acc.LastUpload)
+		return nil
+	default:
```

Comment:
Is there a difference between 1) a server error, 2) a missing account, and 3) an account that hasn't ever uploaded?